scrapy/code/groceries/groceries/spiders/safeway/scraper.py

```diff
@@ -0,0 +1,267 @@
+#! /usr/local/bin/python3
+
+import scrapy
+from scrapy.shell import inspect_response
+from scrapy_splash import SplashRequest
+from scrapy_selenium import SeleniumRequest
+
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from seleniumHelpers import create_parse_request, create_unfiltered_parse_request
+import time
+#TODO change to selenium so that we can update the location!!!
+import re
+
+from util import read_script, convert_cents, store_url, get_next_url, get_url_metadata, lookup_category, finish_url, get_next_pagination, trim_url, convert_to_ounces, convert_units, clean_string
+
+def convert_ppu(incoming_ppu):
```

Comment:
Can you comment methods?
@description
@param
@return (where applicable)

Reply:
Should be good